lib/ui/text.dart

```diff
@@ -2231,17 +2300,22 @@ ByteData _encodeStrut(
     bitmask |= 1 << 4;
     data.setFloat32(byteCount, height, _kFakeHostEndian);
     byteCount += 4;
+    if (leadingDistribution != null) {
+      bitmask |= 1 << 5;
+      data.setInt8(byteCount, leadingDistribution.index);
+      byteCount += 4;
+    }
   }
   if (leading != null) {
-    bitmask |= 1 << 5;
+    bitmask |= 1 << 6;
     data.setFloat32(byteCount, leading, _kFakeHostEndian);
     byteCount += 4;
   }
   if (forceStrutHeight != null) {
-    bitmask |= 1 << 6;
+    bitmask |= 1 << 7;
     // We store this boolean directly in the bitmask since there is
     // extra space in the 16 bit int.
-    bitmask |= (forceStrutHeight ? 1 : 0) << 7;
+    bitmask |= (forceStrutHeight ? 1 : 0) << 8;
   }
 
   data.setInt8(0, bitmask);
```

Comment:
This will need to be expanded to `setInt16` now that there are more than 8 flag bits.

The initialization of `byteCount` at the beginning of `_encodeStrut` will also change to `2` to match the width of the bit mask.

Reply:
squeezed `forceStrutHeight` into 1 bit by treating `null` as false.